### ISSUE TITLE
Select crashed with key of undefined on virtual list

### ISSUE THREAD

**Issue by ThePainnn** [ant-design/ant-design#24329]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://stackblitz.com/edit/react-oave3a?embed=1&file=index.js](https://stackblitz.com/edit/react-oave3a?embed=1&file=index.js)

### Steps to reproduce
 - Type ru then select ru and click back into the select as to make another selection.
 - Keeps on selecting value and typings if first step did not trigger it.

### What is expected?
Not to crash on selected value

### What is actually happening?
Seems to be random. Happen after selection is made and we click back into the Select to make another selection. Seems to happen only for virtualized value (not initially shown when clicking on the dropdown)

Workaround set dropdownMatchSelectWidth to true. **Setting it to false trigger the error**. This is the root property cause.

| Environment | Info |
|---|---|
| antd | 4.2.4 |
| React | 16.12.0 |
| System | Windows 10 64bit |
| Browser | Version 81.0.4044.138 (Official Build) (64-bit) |

---
See https://github.com/ant-design/ant-design/issues/20874 for original post

**Comment by ThePainnn** [comment]
Code use to reproduce: 
```
import React from "react";
import ReactDOM from "react-dom";
import "antd/dist/antd.css";
import "./index.css";
import { Select } from "antd";

const { Option } = Select;

class SearchInput extends React.Component {
  state = {
    data: [],
    value: undefined
  };

  handleChange = value => {
    this.setState({ value });
  };

  render() {
    const languages = [
     "ar", "ar-AE", "ar-BH", "ar-DZ", "ar-EG", "ar-IQ", "ar-JO", "ar-KW", "ar-LB", "ar-LY", "ar-MA", "ar-OM", "ar-QA", "ar-SA", "ar-SD", "ar-SY", "ar-TN", "ar-YE", "be", "be-BY", "bg", "bg-BG", "ca", "ca-ES", "cs", "cs-CZ", "da", "da-DK", "de", "de-AT", "de-CH", "de-DE", "de-LU", "el", "el-CY", "el-GR", "en", "en-AU", "en-CA", "en-GB", "en-HK", "en-IE", "en-IN", "en-MT", "en-NZ", "en-PH", "en-SG", "en-US", "en-ZA", "es", "es-AR", "es-BO", "es-CL", "es-CO", "es-CR", "es-DO", "es-EC", "es-ES", "es-GT", "es-HN", "es-MX", "es-NI", "es-PA", "es-PE", "es-PR", "es-PY", "es-SV", "es-US", "es-UY", "es-VE", "et", "et-EE", "fa", "fa-IR", "fi", "fi-FI", "fr", "fr-BE", "fr-CA", "fr-CH", "fr-FR", "fr-LU", "ga", "ga-IE", "he", "he-IL", "hi-IN", "hr", "hr-HR", "hu", "hu-HU", "id", "id-ID", "is", "is-IS", "it", "it-CH", "it-IT", "ja", "ja-JP", "kk-KZ", "ko", "ko-KR", "lt", "lt-LT", "lv", "lv-LV", "mk", "mk-MK", "ms", "ms-MY", "mt", "mt-MT", "nb", "nb-NO", "nl", "nl-BE", "nl-NL", "nn-NO", "no", "no-NO", "pl", "pl-PL", "pt", "pt-BR", "pt-PT", "ro", "ro-RO", "ru", "ru-KZ", "ru-RU", "se", "se-NO", "sk", "sk-SK", "sl", "sl-SI", "sq", "sq-AL", "sr", "sr-BA", "sr-CS", "sr-ME", "sr-RS", "sv", "sv-SE", "th", "th-TH", "tr", "tr-TR", "uk", "uk-UA", "vi", "vi-VN", "zh", "zh-CN", "zh-HK", "zh-SG", "zh-TW",
    ];

    const options = languages.map(v => {
      return (
        <Option key={v} value={v} title={v}>
          {v}
        </Option>
      );
    });

    return (
      <Select
        showSearch
        allowClear
        mode="select"
        value={this.state.value}
        style={this.props.style}
        dropdownMatchSelectWidth={false}
        onChange={this.handleChange}
      >
        {options}
      </Select>
    );
  }
}

ReactDOM.render(
  <SearchInput placeholder="input search text" style={{ width: 200 }} />,
  document.getElementById("container")
);
```

**Comment by xilihuasi** [comment]
in the latest version of antd@4.2.5, it works normally.Seem to have been fixed

**Comment by afc163** [comment]
@ThePainnn Could you confirm it?

**Comment by ThePainnn** [comment]
Cool I'll update today and let you know. Thanks!

**Comment by ThePainnn** [comment]
Yes I confirm its fixed in 4.2.5. Thanks a lot!
